graphics/minesweeper.py

```diff
@@ -0,0 +1,222 @@
+import pygame as pg
+import random
+
+LEFT_CLICK = 1
+RIGHT_CLICK = 3
+
+
+class Game:
+    """
+    Interactive GUI class with no game logic
+    """
+    def __init__(self, x_tiles, y_tiles, tile_dim=20):
+        """
+        Game session constructor
+        :param x_tiles: Number of tiles in the x dimension
+        :param y_tiles: Number of tiles in the y dimension
+        :param tile_dim: Width/height of the tiles in pixels
+        """
+        self.x_tiles = x_tiles
+        self.y_tiles = y_tiles
+        self.tile_dim = tile_dim
+        pg.init()
+
+        self.board = Board(x_tiles, y_tiles, tile_dim)
+        self.running = True
+
+    def loop(self):
+        """
+        Render the board and handle mouse input
+        :return:
+        """
+        while self.running:
+            for event in pg.event.get():
+                if event.type == pg.QUIT:
+                    self.running = False
+                if event.type == pg.MOUSEBUTTONDOWN:
+                    x, y = event.pos
+                    t_x, t_y = self.board.map_click(x, y)
+                    button = event.button
+
+                    if button == LEFT_CLICK:
+                        (running, _) = self.board.check_tile(t_x, t_y)
+
+                    elif button == RIGHT_CLICK:
+                        self.board.toggle_flag(t_x, t_y)
```

Comment:
I think that instead of doing logic like this in the Board class it should be done in the engine. So call engine.toggle_flag or whatever and then the board just draws it.